### PR TITLE
ui: adding create tag option and generic alerts (PROJQUAY-5290)

### PR DESCRIPTION
**Pull request by bcaton85**
Adds tag options dropdown and the create tag option. Also adds generic alerts through the `useAlerts` hook.

image

### CODE REVIEW

File: web/src/hooks/UseTags.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+export function useTags(org: string, repo: string){
+    const {
+      mutate: mutateCreateTag,
+      isSuccess: successCreateTag,
+      isError: errorCreateTag,
+    } = useMutation(
+      async ({tag, manifest}: {tag: string; manifest: string}) =>
+        createTag(org, repo, tag, manifest),
+      {},
```
Comment (harishsurf) [inline web/src/hooks/UseTags.ts:12]: why is this needed?

File: web/src/resources/TagResource.ts [hunk 2]
```diff
@@ -264,3 +264,10 @@ export async function getSecurityDetails(
   assertHttpCode(response.status, 200);
   return response.data;
 }
+
+export async function createTag(org: string, repo: string, tag: string, manifest: string) {
+    await axios.put(
+      `/api/v1/repository/${org}/${repo}/tag/${tag}`,
+      {manifest_digest: manifest},
+    );
```
Comment (harishsurf) [inline web/src/resources/TagResource.ts:272]: could we also add type checking, assert on status code and try catch in case of error - to keep it consistent with  existing axios requests?

File: web/src/routes/Alerts.tsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import { Alert, AlertActionCloseButton, AlertGroup } from "@patternfly/react-core";
+import { useRecoilState } from "recoil";
+import { alertState } from "src/atoms/AlertState";
+
+
+export default function Alerts(){
+    const [alerts, setAlerts] = useRecoilState(alertState);
+    return (
+    <AlertGroup isToast isLiveRegion>
+        {alerts.map(alert=><Alert
```
Comment (harishsurf) [inline web/src/routes/Alerts.tsx:10]: nit: doesn't seem to be formatted. It would be a good idea to add a format check for .tsx files in our CI

Comment (harishsurf) [inline web/src/routes/Alerts.tsx:10]: what do you think about adding a [timeout](https://www.patternfly.org/v4/components/alert/#alert-timeout) for alerts?

File: web/src/routes/RepositoryDetails/Tags/TagsActionsAddTagModal.tsx [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+            setValue('')
+            props.setIsOpen(false)
+        }
+    }, [errorCreateTag]);
+
+    return (
+        <>
+            <Modal
+                id="add-tag-modal"
+                title={`Add tag to manifest ${props.manifest}`}
```
Comment (harishsurf) [inline web/src/routes/RepositoryDetails/Tags/TagsActionsAddTagModal.tsx:34]: could you make the font smaller so that the `sha` doesn't cut off?  The mocks seem to have smaller font size

**Review by harishsurf — COMMENTED** [review]
some minor comments, otherwise lgtm!

File: web/src/hooks/UseTags.ts (hunk 1 above)
Comment (bcaton85) [reply web/src/hooks/UseTags.ts:12]: It isn't really, just left over. It can be removed.

File: web/src/resources/TagResource.ts (hunk 2 above)
Comment (bcaton85) [reply web/src/resources/TagResource.ts:272]: This is now being handled by react query, we should consider removing that code from the other axios requests.

File: web/src/routes/Alerts.tsx (hunk 3 above)
Comment (bcaton85) [reply web/src/routes/Alerts.tsx:10]: agreed

Comment (bcaton85) [reply web/src/routes/Alerts.tsx:10]: I was thinking we would want to keep an error alert so the user has a chance to debug. It makes sense for successful messages though.

File: web/src/routes/RepositoryDetails/Tags/TagsActionsAddTagModal.tsx (hunk 4 above)
Comment (bcaton85) [reply web/src/routes/RepositoryDetails/Tags/TagsActionsAddTagModal.tsx:34]: done

**Review by harishsurf — APPROVED** [review]
lgtm!

**Review by harishsurf — APPROVED** [review]
